### ISSUE TITLE
Removing experimental status of Matrix Stats

### ISSUE THREAD

**Issue by polyfractal** [elastic/elasticsearch#53065]
Matrix stats was [introduced ~four years ago](https://github.com/elastic/elasticsearch/pull/18300) as an experimental feature... and remains experimental to this day.  We need to determine what is barring it from becoming a GA feature and track progress towards that goal.

The last time we talked, we were wanting to postpone GA status until after the VS refactor in case drastic changes were needed to accommodate the new framework.  But we have since decided to wait on converting multi-VS aggregations because we're not convinced the current abstractions are correct / worth converting, so that blocker is no longer present.

I can't think of another reason that we shouldn't flip the GA switch on `matrix_stats`.  If anyone has a todo or needed change to `matrix_stats` please share.  Otherwise we should make the change sooner than later ;)

**Comment by polyfractal** [comment]
We discussed this in our team meeting a little while ago and can't think of any compelling reason to keep `matrix_stats` experimental.  We're going to open a PR to adjust the docs. :tada:

**Comment by nik9000** [comment]
We've removed experimental in #64318.
